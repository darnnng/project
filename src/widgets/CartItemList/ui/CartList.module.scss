@import './../../../shared/variables.scss';
.cartContainer {
  display: flex;
  flex-direction: column;
  row-gap: 20px;
  margin-top: 20px;
}

.noItemsText {
  font-size: $fontMid;
}
