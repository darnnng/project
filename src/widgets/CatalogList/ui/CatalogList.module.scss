@import './../../../shared/variables.scss';

.itemsContainer {
  display: flex;
  justify-content: center;
  column-gap: 40px;
  row-gap: 50px;
  flex-wrap: wrap;
  font-family: $fontFamily;
}
