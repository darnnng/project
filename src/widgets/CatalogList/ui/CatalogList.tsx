import React from 'react';
import { useParams } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { useAppSelector } from '@shared/model/reduxHooks';
import { selectedFilter } from '@features/Sorting/model/filterSlice';
import { selectedPage } from '@features/Pagination/model/paginationSlice';
import { CatalogItem } from '@entities/catalogItem';
import { useCatalogList } from '../model/useCatalogList';
import { IListItem } from '../model/CatalogList.interface';
import styles from './CatalogList.module.scss';

export const CatalogList = () => {
  const { category } = useParams();
  const { page } = useAppSelector(selectedPage);
  const { filter } = useAppSelector(selectedFilter);
  const { t } = useTranslation();
  const { itemsList } = useCatalogList(
    `https://apidojo-hm-hennes-mauritz-v1.p.rapidapi.com/products/list?country=us&lang=en&currentpage=${page}&pagesize=28&categories=${category}&sortBy=${filter}`,
    page,
    filter
  );
  return (
    <>
      {itemsList.length ? (
        <div className={styles.itemsContainer}>
          {itemsList?.map((item: IListItem) => (
            <CatalogItem key={item.id} item={item} category={category!} />
          ))}
        </div>
      ) : (
        <p className={styles.textNoItems}> {t('Looks like there no items yet...')}</p>
      )}
    </>
  );
};
