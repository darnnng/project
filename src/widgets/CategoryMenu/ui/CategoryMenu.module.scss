@import './../../../shared/variables.scss';
.categoriesWrapper {
  font-family: $fontFamily;
  font-size: $fontMid;
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-top: 50px;
  column-gap: 15px;
  flex-wrap: wrap;
  margin: 50px 10px 0px;
}

.categoryBox {
  &:hover {
    color: $hoverColor;
    cursor: pointer;
  }
}
