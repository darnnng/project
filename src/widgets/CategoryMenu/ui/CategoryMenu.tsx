import React from 'react';
import { useNavigate } from 'react-router-dom';
import { RoutePath } from '@src/shared/constants/routes';
import { useCategoriesList } from '../lib/useCategories';
import { ICategoryItem } from '../model/CategoryMenu.interface';
import styles from './CategoryMenu.module.scss';

export const CategoryMenu = () => {
  const navigate = useNavigate();
  const categoriesList = useCategoriesList();
  const handleClick = (category: string) => {
    navigate(`/${RoutePath.CATALOG}/${category}`);
  };

  return (
    <div className={styles.categoriesWrapper}>
      {categoriesList?.map((category: ICategoryItem) => (
        <div
          className={styles.categoryBox}
          key={category.name}
          onClick={() => handleClick(category.id || category.name)}
        >
          {category.name}
        </div>
      )) || []}
    </div>
  );
};
