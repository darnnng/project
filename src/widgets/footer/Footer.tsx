import React from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { RoutePath } from '@src/shared/constants/routes';
import styles from './Footer.module.scss';

export const Footer = () => {
  const { t } = useTranslation();
  return (
    <footer className={styles.footer}>
      <div className={styles.footerUp}>
        <div className={styles.aboutSection}>
          <p className={styles.footerBigTitle}>{t('ABOUT US')}</p>
          <div className={styles.footerLinks}>
            <p>
              <Link className={styles.footerLinkText} to={`/${RoutePath.CATALOG}`}>
                {t('Our stores')}
              </Link>
            </p>
            <p>
              <Link className={styles.footerLinkText} to={`/${RoutePath.CATALOG}`}>
                {t('Catalog')}
              </Link>
            </p>
          </div>
        </div>
        <div className={styles.socialMedia}>
          <p className={styles.footerBigTitle}>{t('CONTACTS')}</p>
          <div className={styles.footerLinksSocial}>
            <Link to={`https://www.facebook.com/`}>
              <span className={styles.materialSymbolsOutlined}>groups</span>
            </Link>
            <Link to={`https://www.google.com/gmail/about/`}>
              <span className={styles.materialSymbolsOutlined}>mail</span>
            </Link>
          </div>
        </div>
      </div>
      <div className={styles.footerDown}>
        <span>Threads & Co., Inc. All rights reserved</span>
      </div>
    </footer>
  );
};
