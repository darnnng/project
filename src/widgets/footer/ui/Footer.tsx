import { Link } from 'react-router-dom';
import { Trans, useTranslation } from 'react-i18next';
import { linkData, linkDataSocial } from '../model/links';
import styles from './Footer.module.scss';

export const Footer = () => {
  const { t } = useTranslation('translation');
  return (
    <footer className={styles.footer}>
      <div className={styles.footerUp}>
        <div className={styles.aboutSection}>
          <p className={styles.footerBigTitle}>
            <Trans>{t('about')}</Trans>
          </p>
          <div className={styles.footerLinks}>
            {linkData.map((link, index) => (
              <p key={index}>
                <Link className={styles.footerLinkText} to={`/${link.path}`}>
                  {t(link.text)}
                </Link>
              </p>
            ))}
          </div>
        </div>
        <div className={styles.socialMedia}>
          <p className={styles.footerBigTitle}>{t('contacts')}</p>
          <div className={styles.footerLinksSocial}>
            {linkDataSocial.map((link, index) => (
              <Link key={index} to={link.url}>
                <span className={styles.materialSymbolsOutlined}>{link.text}</span>
              </Link>
            ))}
          </div>
        </div>
      </div>
      <div className={styles.footerDown}>
        <span>Threads & Co., Inc. All rights reserved</span>
      </div>
    </footer>
  );
};
