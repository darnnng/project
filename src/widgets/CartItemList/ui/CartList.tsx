import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { currentUser } from '@entities/user/model/userSlice';
import { useAppSelector } from '@shared/model/reduxHooks';
import { useHandleError } from '@shared/model/useHandleError';
import { deleteFromCartDb, getItemsFromCart } from '@features/AddToCart/api/cartApi';
import { CartItem } from '@entities/cartItem/ui/CartItem';
import { ICartItem } from '../model/CartItemList.interface';
import styles from './CartList.module.scss';

export const CartList = () => {
  const { t } = useTranslation();
  const [cartItems, setCartItems] = useState<Record<string, ICartItem>>({});
  const { userId } = useAppSelector(currentUser);
  const handleError = useHandleError();

  useEffect(() => {
    getItemsFromCart(userId!) //TO-DO SORT BY ADDING DATE
      .then((result) => {
        setCartItems(result);
      })
      .catch((error) => {
        handleError(error);
      });
  }, [userId, handleError]);

  const handleDeleteFromCart = (userId: string, item: ICartItem) => {
    deleteFromCartDb(userId!, item)
      .then(() => {
        setCartItems((prevCartItems) => {
          const { [item.id]: deletedItem, ...restOfItems } = prevCartItems;
          return restOfItems as Record<string, ICartItem>;
        });
      })
      .catch((error) => {
        handleError(error);
      });
  };

  return (
    <div className={styles.cartContainer}>
      {Object.values(cartItems).length ? (
        Object.values(cartItems)?.map((item: ICartItem) => (
          <CartItem
            key={item.id}
            item={item}
            handleDeleteFromCart={handleDeleteFromCart}
            userId={userId!}
          />
        ))
      ) : (
        <p className={styles.noItemsText}>{t('Looks like there no items yet...')}</p>
      )}
    </div>
  );
};
