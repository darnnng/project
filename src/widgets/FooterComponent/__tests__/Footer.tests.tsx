import { screen, waitFor } from '@testing-library/react';
import { customRender } from '@shared/lib/utils.tests';
import { Footer } from '../ui/Footer';

describe('Footer test', () => {
  test('renders elements', async () => {
    customRender(<Footer />);
    screen.debug();
    await waitFor(() => {
      expect(screen.getByText(/Catalog/i)).toBeInTheDocument();
    });
    const linkElements = screen.getAllByRole('link');
    expect(linkElements).toHaveLength(4);
  });

  test('footer snapshot matches', async () => {
    await waitFor(() => {
      const { container } = customRender(<Footer />);
      expect(container).toMatchSnapshot();
    });
  });
});
