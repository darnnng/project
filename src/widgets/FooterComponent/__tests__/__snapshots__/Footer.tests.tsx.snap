// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test footer snapshot matches 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="footerUp"
    >
      <div
        class="aboutSection"
      >
        <p
          class="footerBigTitle"
        >
          about
        </p>
        <div
          class="footerLinks"
        >
          <p>
            <a
              class="footerLinkText"
              href="/maps"
            >
              ourstores
            </a>
          </p>
          <p>
            <a
              class="footerLinkText"
              href="/catalog"
            >
              catalog
            </a>
          </p>
        </div>
      </div>
      <div
        class="socialMedia"
      >
        <p
          class="footerBigTitle"
        >
          contacts
        </p>
        <div
          class="footerLinksSocial"
        >
          <a
            href="https://www.facebook.com/"
          >
            <span
              class="materialSymbolsOutlined"
            >
              groups
            </span>
          </a>
          <a
            href="https://www.google.com/gmail/about/"
          >
            <span
              class="materialSymbolsOutlined"
            >
              mail
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="footerDown"
    >
      <span>
        Threads & Co., Inc. All rights reserved
      </span>
    </div>
  </footer>
</div>
`;
